app/views/idv/shared/_step_indicator.html.erb

```diff
@@ -0,0 +1,49 @@
+<%#
+locals:
+* step: Current step.
+* verify_pending: Whether user is awaiting verification by mail. Defaults to `false`.
+%>
+<%
+steps = [
```

Comment:
I know it's a WIP but I'd consider maybe moving this to a constant or putting somewhere else? Then change the `title:` keys to be the i18n slugs only not the translated values

Reply:
> I know it's a WIP but I'd consider maybe moving this to a constant or putting somewhere else? Then change the `title:` keys to be the i18n slugs only not the translated values

I put it up early for this kind of feedback, so it's welcomed 😄 

Is there a pattern you have in mind for where it might make sense to have this kind of constant? A `config/*.yml` ? A helper module like [`AccordionHelper`](https://github.com/18F/identity-idp/blob/main/app/helpers/accordion_helper.rb)?

Maybe there's a way to define or infer it from [`DocAuthFlow`](https://github.com/18F/identity-idp/blob/main/app/services/idv/flows/doc_auth_flow.rb) or each step's implementation 🤔  Then again, not every step of the IAL2 flow is actually part of the `FlowStateMachine` flow.

Reply:
DocAuthFlow would be nice, but good point that it doesn't match the UI steps anymore... maybe we just make a new `ial2_flow.rb` or something

```
# app/services/flow/ial2_flow.rb
module Flow
  module Ial2Flow
    STEP_INDICATOR_STEPS = [] # etc
  end
end
```

Reply:
The `pending` handling here makes for an interesting challenge as well in moving it to a constant.

Maybe we could feed through the partial assigns to a proc? 🤷 

```rb
module Idv
  class StepIndicator
    STEPS = [
      // ...
      {
        slug: :verify_phone_or_address,
        title: 'idv.step_indicator.verify_phone_or_address',
        subtitle: 'idv.step_indicator.pending_confirmation',
        pending: Proc.new { |local_assigns| local_assigns[:verify_pending] },
      },
      // ...
    ]
  end
end
```

```erb
<% Idv::StepIndicator::STEPS.each_with_index do |inner_step, index| %>
  <%= render(
    'shared/step_indicator_step',
    current: step == inner_step[:slug],
    complete: current_step_index > index,
    subtitle: t(inner_step[:subtitle]) if inner_step[:subtitle],
    pending: inner_step[:pending]&.call(local_assigns),
  ) { t(inner_step[:title]) } %>
<% end %>
```

Reply:
mmmm I would not go the proc approach, maybe then we have the flow class act as a builder:

```
# in the controller or step or something outside the template
@ial2_flow = Flows::Ial2Flow.new(verify_pending: true).build
```

and if we did deferred evaluation like that we could do the `t()` inside that `#build` method

Reply:
I've been trying a few different approaches this morning, and pushed up b9ddc6ea93e2e7f70bcdea6a6133dd3c3235eba8 as a work-in-progress. I'm not loving it, but will let it sit and think on it. 

A few pros:

- Integrated into flow, and each step assigns their own step association.
- Automatically prepended to step view, as opposed to littering each step's view.
- Lets exceptional cases like "Verify Pending" be handled only where they're relevant, and where the relevant data is most easily accessible.

And cons:

- Passing around translation keys in general, but more specifically that the step indicator partial is responsible for applying the translation. Ideally it'd just receive the already-translated string. At the very least, I may change this to something like `title: local_assigns[:title] || t(local_assigns[:title_key])`.
- A lot more indirection. Reflecting on the previous implementation, I don't know that I disliked as much that the `idv/shared/step_indicator` partial knew about all the steps in the flow, at least in that it was standalone from the general-purpose `shared/step_indicator` and avoided duplication across step views.

And general thoughts:

- To your original suggestions, it's made challenging by the fact that there's not great points to inject into controller handling for each step, and there's quite a few different controllers for the flow.
- Investing into FlowStateMachine enhancements like this may not be worth the effort if we plan to redo it.

Reply:
I tried to improve this a bit more, mostly in 3af1a3d0b8581f748942ee67644ad8d2e5e1ba42, where translation occurs by virtue of defining all flow labels in a step indicator locale configuration. This has the added advantage of simplifying the flow constant to simply an array of symbols, and avoids duplication between the step name and translated label.